tests/unit-tests/test_decoder.py

```diff
@@ -983,6 +986,378 @@ def test_MLPEFeatEdgeDecoder(h_dim, feat_dim, out_dim, num_ffn_layers):
         pred = out.argmax(dim=1)
         assert_almost_equal(prediction.cpu().numpy(), pred.cpu().numpy())
 
+@pytest.mark.parametrize("in_units", [16, 64])
+@pytest.mark.parametrize("num_classes", [4, 8])
+def test_DenseBiDecoder(in_units, num_classes):
+
+    g, h, edge_type = generate_dummy_constant_graph(in_units)
+
+    with th.no_grad():
+        # We trick the decoder to predict TARGET_CLASS by adding 
+        # INCREMENT_VALUE at TARGET_CLASS' index in the combine_basis weight matrix.
+        # Because basis matrices are set to identity, the output of the bases are all 
+        # equal. So, by modifying the weight of combine_basis, we trick the decoder
+        # to force it to predict TARGET_CLASS. 
+        INCREMENT_VALUE = 100
+        TARGET_CLASS = 2
+        decoder = DenseBiDecoder(
+            in_units=in_units,
+            num_classes=num_classes,
+            multilabel=False,
+            target_etype=edge_type,
+            use_bias=False
+        )
+        decoder.eval()
+
+        # Test bias doesn't exist on combine_basis nn.Linear
+        assert decoder.combine_basis.bias is None
+
+        for i in range(decoder.num_basis):
+            decoder.basis_para[i, :, :] = th.eye(in_units)
+        th.nn.init.eye_(decoder.combine_basis.weight)
+        decoder.combine_basis.weight[TARGET_CLASS][0] += INCREMENT_VALUE # Trick decoder
+
+        prediction = decoder.predict(g, h)
+        assert th.all(prediction == TARGET_CLASS)
+        assert prediction.shape[0] == g.num_edges()
```

Comment:
You may use `prediction.shape = (g.num_edges(), num_classes)` to check all dimensions.

Reply:
If the prediction is a 1D tensor, we can use `prediction.shape[0] = g.num_edges()`.